### PR TITLE
DDF-5909 Clear geo search drawing correctly when using query-basic form

### PR DESCRIPTION
**Pull request by a-asaad**
**Forward port of [codice/ddf#5910](https://github.com/codice/ddf/pull/5910)**

#### What does this PR do?
Clears drawings from a basic location search when toggling from "Somewhere Specific" to "Anywhere" location filters.

#### Who is reviewing it? 
@jordanwilking @andrewkfiedler @stevenmalmgren 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@vinamartin

#### How should this be tested?
On the Basic Search menu, click the "Somewhere Specific" location filter. 
Draw any geo on a map visual.
Switch back to the "Anywhere" filter. The drawing should be cleared.

#### What are the relevant tickets?
Fixes: #5909 

#### Screenshots
Screen Shot 2020-03-17 at 10 13 38 AM

After toggling back to "Anywhere"
Screen Shot 2020-03-17 at 10 14 14 AM

### CODE REVIEW

**Review by andrewkfiedler — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerCollection.controller.js [hunk 1]
```diff
@@ -42,6 +43,7 @@ const Controller = CommonLayerController.extend({
   initialize() {
     // there is no automatic chaining of initialize.
     CommonLayerController.prototype.initialize.apply(this, arguments)
+    this.listenTo(wreqr.vent, 'map:requestRender', this.requestRender)
```
Comment (andrewkfiedler) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerCollection.controller.js]: I think we should move this over to the initialization of the map in: https://github.com/codice/ddf-ui/blob/master/ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js#L103

We'll have to handle unlistening ourselves though there, so we'll have to update the destroy method as well: https://github.com/codice/ddf-ui/blob/master/ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js#L1024

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/property/property.js [hunk 2]
```diff
@@ -108,6 +108,10 @@ module.exports = Backbone.Model.extend({
     currentValue.sort()
     switch (this.getCalculatedType()) {
       case 'location':
+        const currentShape = currentValue[0].mode
```
Comment (andrewkfiedler) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/property/property.js]: Why is this part necessary?

Comment (a-asaad) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/property/property.js]: I wanted to restrict triggering a render to shape changes in the location dropdown only. Otherwise, it was triggering a render everytime the drawing was changed.

**Review by andrewkfiedler — APPROVED** [review]

**Review by bdeining — APPROVED** [review]

**Comment by jlcsmith** [comment]
Hero successful
